src/libsync/discovery.cpp

```diff
@@ -34,6 +37,61 @@ namespace OCC {
 
 Q_LOGGING_CATEGORY(lcDisco, "sync.discovery", QtInfoMsg)
 
+bool ProcessDirectoryJob::checkForInvalidFileName(const SyncFileItemPtr &item)
```

Comment:
this processing should be called from within `ProcessDirectoryJob::process()` where remote discovery is already done such that you can check in `std::map<QString, Entries> entries;` that the remote conflicting entries exist or not
the blocking call to `PropFindJob` will be a problem for performance and are redundant after we have done LsColJob on the directory content itself

Reply:
OKay, I changed it. Way better I think. Thanks for the hint.